### ISSUE TITLE
Incorrect Authentication credentials. Error while running script.

### ISSUE THREAD

**Issue by rs1990** [rghuckins/robinhood-google-sheets#9]
HI  

I followed the instructions and when i try to login , i get this error.

I have changed the username and password to the correct details. i confirmed using the same login credentials and they are correct.

Could you help please

````
Error Robinhood API request failed. 401: {"detail":"Incorrect authentication credentials."} (line 89).
````

**Comment by fkotsian** [comment]
Should be closed by https://github.com/rghuckins/robinhood-google-sheets/pull/5, pls try copy/pastaing latest robinhood.gs into your sheets

(you need a client_id for RH's new Oauth flow 😄 )

**Comment by joecho27** [comment]
I copied the latest robinhood.gs (the one w/ client_id in it) and I still get the same error as OP.

**Comment by ParitoshKelkar** [comment]
Same issue here as well.

**Comment by fkotsian** [comment]
Sorry guys, misspoke. You need a device_id as well. Check out https://github.com/rghuckins/robinhood-google-sheets/pull/10/files

**Comment by ParitoshKelkar** [comment]
Still facing the same error ..

**Comment by philgapp** [comment]
I can confirm I am getting the identical error. I am building out a personal trade management tracker and planner in Sheets and Scripts. The client_id for my personal RH session in a browser matches what is in the latest version of this script:
window.oauthClientId = "c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS"; (BROWSER)
and
'client_id': "c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS", (SCRIPT)
I did have MFA enabled but it is now disabled. I am going to attempt a logout in the browser, cache refresh and logging back in....

**Comment by philgapp** [comment]
> I can confirm I am getting the identical error. I am building out a personal trade management tracker and planner in Sheets and Scripts. The client_id for my personal RH session in a browser matches what is in the latest version of this script:
> window.oauthClientId = "c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS"; (BROWSER)
> and
> 'client_id': "c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS", (SCRIPT)
> I did have MFA enabled but it is now disabled. I am going to attempt a logout in the browser, cache refresh and logging back in....

This is resolved for me by adding **device_token** to the function below: 
function getAccessToken_() {
  var url = robinhoodApiBaseUrl + '/oauth2/token/';
  var payload = {
    'grant_type': "password",
    'scope': "internal",
    'client_id': "c82SH0WZOsabOXGP2sxqcj34FxkvfnWRZBKlBjFS",
    'device_token' : "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx",
    //'auth_token' : "4e4a67a2b3cbdd369a3ee5963becdb776073518a",
    'username': robinhoodUsername,
    'password': robinhoodPassword
  };

You can find this value by logging in with Chrome, Inspecting the page, go to Applications and expand the Cookies on the left, click the one for robinhood.com and then filter on the right for device_id. Copy that value and add it as device_token as above, and save and try again. Let me know, I want to actively use this in my project and share for help, as I'm not an expert developer by any means. I am however passionate about more intelligent trading software that helps educate and guide new investors.

**Comment by ParitoshKelkar** [comment]
That worked for me too, thanks @philgapp

**Comment by KSanchez999** [comment]
I'm having this issue as well. I'm using the robinhood.gs script updated 10 days ago. I have a device token but I'm still getting an authentication credentials error.

**Comment by mrjayenem** [comment]
@rghuckins I have all the correct info but I still receive this same error. Any help?

**Comment by adeedchoudhury** [comment]
Yep facing same issue as 2 commentors above me - I have the device_token (from the stackoverflow link) but still getting a 401 error code... (Invalid authentication credentials)

**Comment by cstampar** [comment]
Also got my device token and put in script editor, i get a text with a verification code and the same response as others in sheets cell: "incorrect authentication credentials"

**Comment by dheerajn** [comment]
Thank you for this awesome code but I have the same issue. I took the device_id value from the website and pasted it in the code and saved but I still have issue.
Can someone help me please?
I have getting the following error
```
Robinhood API request failed. 401: {"detail":"Incorrect authentication credentials."} (line 84).
```
